os/mm/umm_heap/umm_malloc.c

```diff
@@ -85,6 +85,33 @@
  * Public Functions
  ****************************************************************************/
 
+/************************************************************************
+ * Name: malloc_at
+ *
+ * Description:
+ *   Allocate memory to the specific heap.
+ *
+ * Parameters:
+ *   heap_index - Index of specific heap.
+ *   size - Size (in bytes) of the memory region to be allocated.
+ *
+ * Return Value:
+ *   The address of the allocated memory (NULL on failure to allocate)
+ *
+ ************************************************************************/
+
+#if CONFIG_MM_NHEAPS > 1
+void *malloc_at(int heap_index, size_t size)
+{
+#ifdef CONFIG_DEBUG_MM_HEAPINFO
+	ARCH_GET_RET_ADDRESS
+	return mm_malloc(&g_mmheaps[heap_index], size, retaddr);
```

Comment:
I think you need to check heap_index to avoid assert.

Reply:
i feel Similarly Size argument limits validation also needed.
What is your opinion ?

Reply:
size will be checked in mm_malloc.